api/src/api/users/user_routes.py

```diff
@@ -184,3 +186,31 @@ def user_save_opportunity(
     )
 
     return response.ApiResponse(message="Success")
+
+
+@user_blueprint.delete("/<uuid:user_id>/saved-opportunities/<int:opportunity_id>")
+@user_blueprint.output(UserDeleteSavedOpportunityResponseSchema)
+@user_blueprint.doc(responses=[200, 401, 404])
+@user_blueprint.auth_required(api_jwt_auth)
+@flask_db.with_db_session()
+def user_delete_saved_opportunity(
+    db_session: db.Session, user_id: UUID, opportunity_id: int
+) -> response.ApiResponse:
+    logger.info("DELETE /v1/users/:user_id/saved-opportunities/:opportunity_id")
+
+    user_token_session: UserTokenSession = api_jwt_auth.current_user  # type: ignore
+
+    # Verify the authenticated user matches the requested user_id
+    if user_token_session.user_id != user_id:
+        raise_flask_error(401, "Unauthorized user")
+
+    with db_session.begin():
+        # Try to find and delete the saved opportunity
+        result = delete_saved_opportunity(db_session, user_id, opportunity_id)
+
+    if result == 0:
+        raise_flask_error(404, "Saved opportunity not found")
+
+    db_session.commit()
```

Comment:
This doesn't do anything as a commit automatically occurred at the end of the block with `begin()`
```suggestion

```